libcasr/src/rust.rs

```diff
@@ -26,3 +29,283 @@ impl Exception for RustPanic {
         )))
     }
 }
+
+/// Structure provides an interface for processing stacktraces from RUST_BACKTRACE.
+pub struct RustStacktrace;
+
+impl ParseStacktrace for RustStacktrace {
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>> {
+        let re = Regex::new(r"( *\d+: *?0x[0-9a-f]+ - .*\n(?: *at .*\n)?)").unwrap();
+        let mut stacktrace: Vec<_> = re
+            .find_iter(stream)
+            .map(|s| s.as_str().trim().replace('\n', " "))
+            .collect();
+
+        if stacktrace.is_empty() {
+            return Err(Error::Casr(
+                "Couldn't find stacktrace entries in Rust panic output".to_string(),
+            ));
+        }
+
+        // Filter out multiple spaces
+        stacktrace = stacktrace
+            .iter()
+            .map(|e| {
+                e.split(' ')
+                    .filter(|s| !s.is_empty())
+                    .collect::<Vec<_>>()
+                    .join(" ")
+            })
+            .collect();
+
+        Ok(stacktrace)
+    }
+
+    fn parse_stacktrace_entry(entry: &str) -> Result<StacktraceEntry> {
+        let re = Regex::new(r#" *[0-9]+: +0x([0-9a-f]+) - (.+?)(?: *at (.+):([0-9]+):([0-9]+)|$)"#)
```

Comment:
are there always both line number and column?

Reply:
We allowed column to be absent